[Feature](parallel-result-sink) support async fetch from multiple backends concurrently

BiteTheDDDDt opened the pull request:
### What problem does this PR solve?
support async fetch from multiple backends concurrently
If we fetch data one by one with the backend in order, the logical deadlock may be triggered due to memory control, and the load on the backend is also unbalanced.

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [x] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [x] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [x] No.
    - [ ] Yes. <!-- Add document PR link here. eg: https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR should merge into -->

Commit:
support async fetch from multiple backends concurrently

format

format

fix status

fix

fix

fix

fix

fix

fix

fix

fix

fix

update

update

update

update

fix

hello-stephen commented:
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

morningman commented on fe/fe-core/src/main/java/org/apache/doris/qe/ResultReceiverConsumer.java:
```diff
@@ -0,0 +1,124 @@
+    BlockingQueue<Integer> readyOffsets;
+    int finishedReceivers = 0;
+
+    public ResultReceiverConsumer(List<ResultReceiver> resultReceivers) {
```
I suggest to write a ut for this class
Here is a example written by cursor, I didn't not run it, but may help you
[ResultReceiverConsumerTest.java.txt](https://github.com/user-attachments/files/18870715/ResultReceiverConsumerTest.java.txt)

Commit:
update

update

fix checkstyl

fix

More accurate display of predicates on storage layer

Revert "More accurate display of predicates on storage layer"

This reverts commit 67a9c2c6b46b0685df5d7fa196311cecea9098f6.

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

Commit:
update case

disable spill

update

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

Commit:
add more case

BiteTheDDDDt commented:
run buildall

Commit:
add ut

BiteTheDDDDt commented:
run buildall

Commit:
format

BiteTheDDDDt commented:
run buildall

Commit:
update case

Commit:
remove big groovy

BiteTheDDDDt commented:
run buildall

Commit:
fixmat

Commit:
format

BiteTheDDDDt commented:
run buildall
